core/common/src/main/java/alluxio/master/MasterInquireClient.java

```diff
@@ -97,7 +99,8 @@ public static MasterInquireClient createForJobMaster(AlluxioConfiguration conf)
         return ZkMasterInquireClient.getClient(conf.get(PropertyKey.ZOOKEEPER_ADDRESS),
             conf.get(PropertyKey.ZOOKEEPER_JOB_ELECTION_PATH),
             conf.get(PropertyKey.ZOOKEEPER_JOB_LEADER_PATH),
-            conf.getInt(PropertyKey.ZOOKEEPER_LEADER_INQUIRY_RETRY_COUNT));
+            conf.getInt(PropertyKey.ZOOKEEPER_LEADER_INQUIRY_RETRY_COUNT),
+            conf.getEnum(PropertyKey.SECURITY_AUTHENTICATION_TYPE, AuthType.class));
```

Comment:
does this mean the auth type is now a required client side property? i.e. we cannot depend on cluster defaults handshake to determine the auth type.

Reply:
I don't think so. All creators of `MasterInquireClient` use cluster conf directly or indirectly.

Reply:
Doesn't sound OK though, given it's the inquire client. I'll need to test this.

Reply:
Actually, I'm kind of sure there is a path that calls this before fetching configs.

One solution would be to disable zk client authentication only when instructed by a config property. So authentication could disabled only when it's explicitly required by specific deployments. Sent an iteration for implementing it this way.

Reply:
sounds good